src/main/java/org/jabref/model/entry/Date.java

```diff
@@ -65,6 +66,11 @@ public class Date {
                                                    DateTimeFormatterBuilder::appendOptional,
                                                    (builder, formatterBuilder) -> builder.append(formatterBuilder.toFormatter()))
                                            .toFormatter(Locale.US);
+
+        DATE_REGEX = "\\d{4}-\\d{1,2}-\\d{1,2}" + // covers YYYY-MM-DD, YYYY-M-DD, YYYY-MM-D, YYYY-M-D
+                "|\\d{4}\\.\\d{1,2}\\.\\d{1,2}|" + // covers YYYY.MM.DD, YYYY.M.DD, YYYY.MM.D, YYYY.M.D
+                "(January|February|March|April|May|June|July|August|September|" +
+                "October|November|December) \\d{1,2}, \\d{4}"; // covers Month DD, YYYY & Month D, YYYY
```

Comment:
Please comment, why this is not just following code?

Your tests did not check for 2022-1-2. -- Thus, the regesx checking for a one-digit month in (wrong, but close) ISO format, are not that important.

I would really, really, really move the string constant from lines 143 to 152 to `DATE_REGEX` and modify lines 143 to 152 to use `DATE_REGEX`

```java
"\\d{4} / \\d{4}|" + // uuuu / uuuu
              "\\d{4}-\\d{2} / \\d{4}-\\d{2}|" + // uuuu-mm / uuuu-mm
              "\\d{4}-\\d{2}-\\d{2} / \\d{4}-\\d{2}-\\d{2}|" + // uuuu-mm-dd / uuuu-mm-dd
              "(?i)(January|February|March|April|May|June|July|August|September|October|November|December)" +
              "( |\\-)(\\d{1,4}) / (January|February|March|April|May|June|July|August|September|October|November" +
              "|December)( |\\-)(\\d{1,4})(?i-)|" + // January 2015/January 2015
              "(?i)(\\d{1,2})( )(January|February|March|April|May|June|July|August|September|October|November|December)" +
              "( |\\-)(\\d{1,4}) / (\\d{1,2})( )" +
              "(January|February|March|April|May|June|July|August|September|October|November|December)" +
              "( |\\-)(\\d{1,4})(?i-)" // 20 January 2015/20 January 2015
```

Reply:
@koppor 
> Your tests did not check for 2022-1-2.

I took into consideration cases like `YYYY-M-D` in `URLCleanupTest` and also tested manually cases like `YYYY-M-D` and work fine.

I cannot understand why the proposed regex could be useful for a `url-date`. A `url-date` refers to a **specific** date a user accessed a URL address, while lines `143-152` consider date duration cases and also:
- Do not cover cases like `YYYY.MM.DD` or `YYYY-MM-D` or `YYYY-M-D` etc.
- Cover cases like `20 January 2015/20 January 2015` that state duration and not correspond to a url-date.
- Another example of date duration covered `January 2015/January 2015` --> not related to our case in my opinion
- Do not cover cases like `January 2015, 2021` --> a case that a user may consider when recording dates

So may I move lines `143-152` under `DATE_REGEX` but modify them accordingly? Or do I replace the whole current `DATE_REGEX` regex with no further modifications?

I completely understand that the `DATE_REGEX` may be reused by other classes than `URLCleanup` and must follow a certain approach.

What are your thoughts?

Reply:
I read the code now in my IDE. Please revert extracting the constant. Please use make `org.jabref.model.entry.Date#SIMPLE_DATE_FORMATS` public. 

You can then use

                TemporalAccessor parsedDate = SIMPLE_DATE_FORMATS.parse(dateString);

to parse the date

Reply:
Ah, that doesn't work, because it's not a regex.. OK, I need to think closer...